### PR TITLE
[filebeat][metricbeat] Adds extrArgs

### LINKED ISSUE #1421: *beats add ability to specify additional args to the commands being executed
**Describe the feature:**

Add the ability to specify an additional list of arguments to pass to the beats applications, mainly to enable debug mode and levels but can have other uses.

**Describe a specific use case for the feature:**

* Debugging pipelines
* Debugging autodiscover functionality

Example diff for metricbeat:
```
diff --git a/metricbeat/templates/daemonset.yaml b/metricbeat/templates/daemonset.yaml
index c9d253eb..9431c71e 100644
--- a/metricbeat/templates/daemonset.yaml
+++ b/metricbeat/templates/daemonset.yaml
@@ -122,6 +122,9 @@ spec:
         - "-E"
         - "http.enabled=true"
         - "--system.hostfs=/hostfs"
+      {{- range .Values.daemonset.extraArgs }}
+        - {{ . | quote }}
+      {{- end }}
         livenessProbe:
 {{ toYaml .Values.livenessProbe | indent 10 }}
         readinessProbe:
diff --git a/metricbeat/templates/deployment.yaml b/metricbeat/templates/deployment.yaml
index 740c1b7b..50ef169f 100644
--- a/metricbeat/templates/deployment.yaml
+++ b/metricbeat/templates/deployment.yaml
@@ -106,6 +106,11 @@ spec:
           - "-e"
           - "-E"
           - "http.enabled=true"
+        {{- range .Values.deployment.extraArgs }}
+          - {{ . | quote }}
+        {{- end }}
         livenessProbe:
 {{ toYaml .Values.livenessProbe | indent 10 }}
         readinessProbe:
diff --git a/metricbeat/values.yaml b/metricbeat/values.yaml
index 6d01ae07..663d64a8 100755
--- a/metricbeat/values.yaml
+++ b/metricbeat/values.yaml
@@ -31,6 +31,9 @@ daemonset:
   # - name: extras
   #   mountPath: /usr/share/extras
   #   readOnly: true
+  extraArgs: []
+  # - --d
+  # - "*"
   hostAliases: []
   #- ip: "127.0.0.1"
   #  hostnames:
@@ -146,6 +149,9 @@ deployment:
   #   mountPath: /usr/share/extras
   #   readOnly: true
   # such as metricbeat.yml for deployment
+  extraArgs: []
+  # - --d
+  # - "*"
   hostAliases: []
   #- ip: "127.0.0.1"
   #  hostnames:
```

### PR DESCRIPTION
**Pull request by framsouza**
Fix #1421

### CODE REVIEW

File: filebeat/values.yaml [hunk 1]
```diff
@@ -83,7 +87,7 @@ deployment:
   labels: {}
   affinity: {}
   # Include the deployment
-  enabled: false
+  enabled: true
```
Comment (jmlrt) [inline filebeat/values.yaml:90]: this doesn't seem related to the `extraArgs` change

File: metricbeat/requirements.lock [hunk 2]
```diff
@@ -2,5 +2,5 @@ dependencies:
 - name: kube-state-metrics
   repository: https://charts.helm.sh/stable
   version: 2.4.1
-digest: sha256:948dca129bc7c16b138ed8bcbdf666c324d812e43af59d475b8bb74a53e99778
-generated: "2020-10-30T18:58:57.381827+01:00"
+digest: sha256:2d4e8265cb60d5c0f613217b35ab18846969c86a1a9fdf77cfa75c291f18934a
```
Comment (jmlrt) [inline metricbeat/requirements.lock:5]: this doesn't seem related to the extraArgs change

**Review by jmlrt — CHANGES_REQUESTED** [review]
Some unrelated changes to cleans.

Also, could you update the values descriptions in the README files and add some Python tests?

File: metricbeat/requirements.lock (hunk 2 above)
Comment (framsouza) [reply metricbeat/requirements.lock:5]: @jmlrt put the original file back, not quite sure what changed it

File: filebeat/values.yaml (hunk 1 above)
Comment (framsouza) [reply filebeat/values.yaml:90]: fixed

Comment (jmlrt) [reply filebeat/values.yaml:90]: I'm still seeing this param being changed from false to true in this PR.

**Review by jmlrt — CHANGES_REQUESTED** [review]
just need to change back the `deployment.enabled` value, the rest LGTM if tests are OK

**Comment by c4m4** [comment]
We need this
